biojava-structure/src/main/java/org/biojava/nbio/structure/URLIdentifier.java

```diff
@@ -122,7 +122,11 @@ public SubstructureIdentifier toCanonical() throws StructureException{
 			String path = url.getPath();
 			pdbId = guessPDBID(path.substring(path.lastIndexOf("/") + 1));
 		}
-		return new SubstructureIdentifier(new PdbId(pdbId), ranges);
+		PdbId pdbIdObj = null;
+		if (pdbId != null) {
```

Comment:
Good point, now pushed

Reply:
I guess should it be `(PdbId)null` not just `null`.
Otherwise, the function call would be **ambiguous** and the compiler would either 1) throw a compile time exception, or 2) go to the first constructor that matches the given signature (`SubstructureIdentifier(String pdbId, List<ResidueRange> ranges)` in our case) which will try to create a PdbId object out of the passed-in `null` parameter.